api.lisp

```diff
@@ -0,0 +1,77 @@
+;;;; -*- Mode: Lisp; indent-tabs-mode: nil -*-
+
+(in-package :split-sequence)
+
+(declaim (inline list-long-enough-p check-bounds check-tests))
+
+(defun list-long-enough-p (list length)
+  (or (zerop length)
+      (not (null (nthcdr (1- length) list)))))
+
+(defun check-bounds (sequence start end)
+  (progn
+    (check-type start unsigned-byte "a non-negative integer")
+    (check-type end (or null unsigned-byte) "a non-negative integer or NIL")
+    (typecase sequence
+      (list
+       (when end
+         (unless (list-long-enough-p sequence end)
+           (error "The list is too short: END was ~S but the list is ~S elements long."
+                  end (length sequence)))))
+      (t
+       (let ((length (length sequence)))
+         (unless end (setf end length))
+         (unless (<= start end length)
+           (error "Wrong sequence bounds. START: ~S END: ~S" start end)))))))
+
+(defun check-tests (test test-p test-not test-not-p)
+  (when (and test test-p test-not test-not-p)
+    (error "Cannot specify both TEST and TEST-NOT."))
+  (when (and test-not-p (null test-not) (not test-p))
+    (error 'type-error :datum test :expected-type '(or function (and symbol (not null)))))
+  (when (and test-p (null test) (not test-not-p))
```

Comment:
This is up to your choice, but i would recommend to replace this function with a macro and use check-type isntead (it provides interactive restart allowing to set the value correctly).

Reply:
`CHECK-TYPE` will give us the restarts, yes. I'll do that.